lib/services/consumed/status-bar/status-bar.js

```diff
@@ -35,10 +35,10 @@ export function addStatusBar(
 function showKernelCommands(handleKernelCommand, { kernel, markers }) {
   if (!this.signalListView) {
     this.signalListView = new SignalListView(kernel);
-    this.signalListView.onConfirmed = (kernelCommand: { command: string }) =>
-      handleKernelCommand(kernelCommand, { kernel, markers });
   } else {
     this.signalListView.kernel = kernel;
   }
+  this.signalListView.onConfirmed = (kernelCommand: { command: string }) =>
+    handleKernelCommand(kernelCommand, { kernel, markers });
```

Comment:
The issue (in my mind, I havent experimented) is that we want `kernel` to refer to `store.kernel` (essentially the active kernel). If we de-reference the kernel from `store` inside a function, that kernel reference never changes so it would still be the kernel passed in initially even if `store.kernel` changes.

A way to handle this is to pass `store` as an argument instead of de-referencing `{ kernel }` from store whenever this function gets initialized.